mca/src/agent/model_structure.rs

```diff
@@ -11,19 +11,21 @@ pub struct IrisData {
 
 // Model1Trait 특성 정의
 pub trait Model1Trait {
-    async fn test(&self, input: &str) -> Result<String>;
+    // async fn test(&self, input: String) -> Result<String>;
+    fn test(&self, input: String) -> impl std::future::Future<Output = Result<String>> + Send;
 }
 
+
 // Model1 구조체 정의
 pub struct Model1Struct<'a>{
     pub proxy: Model1Proxy<'a>,
 }
 
 // AIModelTrait을 Model1과 Model2에 대한 특성으로 구현
 impl Model1Trait for Model1Struct<'_> {
-    async fn test(&self, input: &str) -> Result<String> {
+    async fn test(&self, input: String) -> Result<String> {
         println!("{:?}", input);
-        let _reply = self.proxy.predict(input).await?;
+        let _reply = self.proxy.predict(&input).await?;
```

Comment:
What is benefit of using `String` instead of `str`?

Reply:
Actually, it is one of the things I tried to solve the Sync error.
In my opinion, it is not important. it is just different to pass call by reference or call by value, I guess.